scripts/tasks.py

```diff
@@ -509,3 +510,242 @@ def timetool_correct(config):
+    batch_size = task.batch_size
+    filename = task.filename
+    tag = task.tag
+    path = task.path
+    overwrite = True
+
+    filename_with_tag = f"{path}{filename}_{tag}.h5"
+
+    if overwrite and os.path.exists(filename_with_tag):
+        os.remove(filename_with_tag)
+
+    # Initialize U, S, mu and var matrices
+    if os.path.exists(filename_with_tag):
+        with h5py.File(filename_with_tag, 'r') as f:
+            previous_U = np.asarray(f.get('U'))
+            previous_S = np.asarray(f.get('S'))
+            previous_mu_tot = np.asarray(f.get('mu'))
+            previous_var_tot = np.asarray(f.get('total_variance'))
+    else:
+        previous_U = None
+        previous_S = None
+        previous_mu_tot = None
+        previous_var_tot = None 
+
+    # Check values
+    print(f"U: {previous_U}, type(U): {type(previous_U)}")
+    print(f"S: {previous_S}, type(S): {type(previous_S)}")
+    print(f"mu: {previous_mu_tot}, type(mu): {type(previous_mu_tot)}")
+    print(f"total_variance: {previous_var_tot}, type(total_variance): {type(previous_var_tot)}")
+
+    # Initialize a list to store the execution times and frobenius norms
+    execution_times = []
+    compression_loss_values = []
+
+    # Initialize lists to store the L2 norms
+    norm_diff_U_list = []
+    norm_diff_S_list = []
+    norm_diff_mu_list = []
+    norm_diff_var_list = []
+
+    #Initialize number of runs
+
+    # list_runs = get_runs(tag)
+    max_run = task.max_run
+    start_run = run
+
+    # Iterate through runs 
+    for run in range(start_run, start_run + max_run):
+
+        start_time = time.time()  
+        
+        # Create a PiPCA instance for the current run
+        pipca = PiPCA(
+            exp=exp,
+            run=run,
+            det_type=det_type,
+            num_images=num_images,
+            num_components=num_components,
+            batch_size=batch_size,
+            priming=False,
+            downsample=False,
+            bin_factor=2,
+            filename = filename_with_tag
+        )
+
+        # Run iPCA for the current run
+        pipca.run_model_full(previous_U, previous_S, previous_mu_tot, previous_var_tot)
+        
+        end_time = time.time()  # Record the end time
+        execution_time = end_time - start_time  # Calculate the execution time
+        execution_times.append(execution_time)  # Append the execution time to the list
+
+        U = pipca.gather_U()
+        S = pipca.S
+        mu_tot = pipca.gather_mu()
+        var_tot = pipca.gather_var()
+
+        # Compute norms change
+        diff_U, diff_S, diff_mu, diff_var = (U - previous_U, S - previous_S, mu_tot - previous_mu_tot, var_tot - previous_var_tot) if previous_U is not None else (U, S, mu_tot, var_tot)
+        norm_diff_U_list.append(np.linalg.norm(diff_U)); norm_diff_S_list.append(np.linalg.norm(diff_S)); norm_diff_mu_list.append(np.linalg.norm(diff_mu)); norm_diff_var_list.append(np.linalg.norm(diff_var))
```

Comment:
I think it's OK to "decompactify" these two lines, to increase readability. Maybe consider creating a function for this?

Reply:
Ok! I did this instead:

```
# Compute norms change
        def compute_norm_difference(current, previous=None):
            if previous is not None:
                diff = current - previous
            else:
                diff = current
            return diff

        # Compute norm differences for each list
        diff_U = compute_norm_difference(U, previous_U)
        diff_S = compute_norm_difference(S, previous_S)
        diff_mu = compute_norm_difference(mu_tot, previous_mu_tot)
        diff_var = compute_norm_difference(var_tot, previous_var_tot)

        # Append results to respective lists
        norm_diff_U_list.append(np.linalg.norm(diff_U))
        norm_diff_S_list.append(np.linalg.norm(diff_S))
        norm_diff_mu_list.append(np.linalg.norm(diff_mu))
        norm_diff_var_list.append(np.linalg.norm(diff_var))

        # Update U and S for the next run
        previous_U = U
        previous_S = S
        previous_mu_tot = mu_tot
        previous_var_tot = var_tot
```